### PR TITLE
Added Makefile

### PR DESCRIPTION
**Pull request by n0toose**
The Makefile is not 100% functional in its current state due to the state of the codebase, but I believe that it's still a good initial implementation.

### CODE REVIEW

File: Sources/Makefile [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+#		you can simply specify XXX for the library. (e.g. the entry for
+#		"libtracker.so" would be "tracker")
+#
+#	-	for GCC-independent linking of standard C++ libraries, you can use
+#		$(STDCPPLIBS) instead of the raw "stdc++[.r4] [supc++]" library names.
+#
+#	- 	if your library does not follow the standard library naming scheme,
+#		you need to specify the path to the library and it's name.
+#		(e.g. for mylib.a, specify "mylib.a" or "path/mylib.a")
+LIBS =  /boot/system/develop/lib/libbe.so \
```
Comment (diversys) [inline Sources/Makefile]: you can replace system `libbe.so` with just `be`

File: Sources/Makefile [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+ /boot/system/lib/libsupc++.so
```
Comment (diversys) [inline Sources/Makefile]: Is it really needed?

File: Sources/Makefile [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+#	or libXXX.a naming scheme. You can specify full paths or paths relative
+#	to the Makefile. The paths included are not parsed recursively, so
+#	include all of the paths where libraries must be found. Directories where
+#	source files were specified are	automatically included.
+LIBPATHS = 
+
+#	Additional paths to look for system headers. These use the form
+#	"#include <header>". Directories that contain the files in SRCS are
+#	NOT auto-included here.
+SYSTEM_INCLUDE_PATHS =  /boot/system/develop/headers/be \
```
Comment (diversys) [inline Sources/Makefile]: I though this should be included by default, maybe other too?

File: Sources/Makefile [hunk 4]
```diff
@@ -0,0 +1,130 @@
+## Haiku Generic Makefile v2.6 ##
+
+## Fill in this file to specify the project being created, and the referenced
+## Makefile-Engine will do all of the hard work for you. This handles any
+## architecture of Haiku.
+
+# The name of the binary.
+NAME = BeApp
```
Comment (diversys) [inline Sources/Makefile]: Replace with GIGOcalc.

**Comment by scottmc** [comment]
@alwayslivid can you address the pending requests on this one?

**Comment by n0toose** [comment]
> @alwayslivid can you address the pending requests on this one?

Will do briefly, missed that one, excuse me for that one.

File: Sources/Makefile [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+#		you can simply specify XXX for the library. (e.g. the entry for
+#		"libtracker.so" would be "tracker")
+#
+#	-	for GCC-independent linking of standard C++ libraries, you can use
+#		$(STDCPPLIBS) instead of the raw "stdc++[.r4] [supc++]" library names.
+#
+#	- 	if your library does not follow the standard library naming scheme,
+#		you need to specify the path to the library and it's name.
+#		(e.g. for mylib.a, specify "mylib.a" or "path/mylib.a")
+LIBS =  be $(STDCPPLIBS)
```
Comment (humdingerb) [inline Sources/Makefile]: Missing ZLayout-1-0.so here

**Comment by humdingerb** [comment]
Thanks!

**Pull request merged**